agent/SlapdConfigAgent.cc

```diff
@@ -422,11 +434,38 @@ YCPValue SlapdConfigAgent::Execute( const YCPPath &path,
             attrs.add("contextCSN");
             LDAPSearchResults *sr = m_lc->search( "cn=config", LDAPConnection::SEARCH_SUB,
                                                   "objectclass=*", attrs );
+            std::vector<LDAPEntry> searchResult;
+            while (LDAPEntry *e = sr->getNext())
+            {
+                searchResult.push_back(LDAPEntry(*e));
+            }
+            // Explicitly load syncprov.so and DB drivers if modules should be loaded explicitly
+            OlcModuleListEntry moduleListEntry;
+            if (moduleLoadPath != "")
+            {
+                for (std::vector<LDAPEntry>::iterator it = searchResult.begin(); it < searchResult.end(); it++)
+                {
+                    // Look for OlcModuleList and remove its LDAPEntry from result
```

Comment:
regarding What vs Why: The comment is
> // Look for OlcModuleList and remove its LDAPEntry from result

But when I read lines 446-454, it is obvious enough that it is looking for a particular element and removing it:
The question is not Why look for it (because: it is used after the loop) but Why remove it?